daemon/CryptoImpl/cryptopluginfunctionwrappers.cpp

```diff
@@ -0,0 +1,305 @@
+                data, std::get<0>(options), std::get<1>(options), &digest);
+    return DataResult(result, digest);
+}
+
+DataResult CryptoPluginWrapper::sign(
+        CryptoPlugin *plugin,
+        const QByteArray &data,
+        const Key &key,
+        std::tuple<Sailfish::Crypto::CryptoManager::SignaturePadding,
+                   Sailfish::Crypto::CryptoManager::DigestFunction> options)
+{
+    QByteArray signature;
+    Result result = plugin->sign(
+                data, key, std::get<0>(options), std::get<1>(options), &signature);
+    return DataResult(result, signature);
+}
+
+ValidatedResult CryptoPluginWrapper::verify(
+        CryptoPlugin *plugin,
+        const QByteArray &signature,
+        const QByteArray &data,
+        const Key &key,
+        std::tuple<Sailfish::Crypto::CryptoManager::SignaturePadding,
+                   Sailfish::Crypto::CryptoManager::DigestFunction> options)
+{
+    bool verified = false;
+    Result result = plugin->verify(
+                signature, data, key, std::get<0>(options), std::get<1>(options), &verified);
+    return ValidatedResult(result, verified);
+}
+
+TagDataResult CryptoPluginWrapper::encrypt(
+        CryptoPlugin *plugin,
+        std::tuple<QByteArray, QByteArray> dataAndIv,
+        const Key &key,
+        std::tuple<Sailfish::Crypto::CryptoManager::BlockMode,
+                   Sailfish::Crypto::CryptoManager::EncryptionPadding> options,
+        const QByteArray &authenticationData)
+{
+    QByteArray ciphertext;
+    QByteArray authenticationTag;
+    Result result = plugin->encrypt(
+                std::get<0>(dataAndIv), std::get<1>(dataAndIv),
+                key,
+                std::get<0>(options), std::get<1>(options),
+                authenticationData,
+                &ciphertext, &authenticationTag);
+    return TagDataResult(result, ciphertext, authenticationTag);
+}
+
+VerifiedDataResult CryptoPluginWrapper::decrypt(
+        CryptoPlugin *plugin,
+        std::tuple<QByteArray, QByteArray> dataAndIv,
+        const Key &key, // or keyreference, i.e. Key(keyName)
+        std::tuple<Sailfish::Crypto::CryptoManager::BlockMode,
+                   Sailfish::Crypto::CryptoManager::EncryptionPadding> options,
+        std::tuple<QByteArray, QByteArray> authDataAndTag)
+{
+    QByteArray plaintext;
+    bool verified = false;
+    Result result = plugin->decrypt(
+                std::get<0>(dataAndIv), std::get<1>(dataAndIv),
+                key,
+                std::get<0>(options), std::get<1>(options),
+                std::get<0>(authDataAndTag), std::get<1>(authDataAndTag),
+                &plaintext, &verified);
+    return VerifiedDataResult(result, plaintext, verified);
+}
+
+CipherSessionTokenResult CryptoPluginWrapper::initialiseCipherSession(
+        CryptoPlugin *plugin,
+        quint64 clientId,
+        const QByteArray &iv,
+        const Key &key, // or keyreference, i.e. Key(keyName)
+        std::tuple<CryptoManager::Operation,
```

Comment:
Could we use structs?

Reply:
... And everywhere in the code - there are a lot of tuples.

Reply:
Ok, I will update to use structs.